editor/scene/gui/container_editor_plugin.cpp

```diff
@@ -0,0 +1,290 @@
+}
+
+void ContainerEditorPlugin::forward_canvas_draw_over_viewport(Control *p_viewport_control) {
+	if (!container || !container->is_visible_in_tree()) {
+		return;
+	}
+
+	Transform2D xform = CanvasItemEditor::get_singleton()->get_canvas_transform() * container->get_screen_transform();
+
+	// NOTE: This color is copied from Camera2DEditor::forward_canvas_draw_over_viewport.
+	// We may want to unify them somehow in the future.
+	Color border_color = Color(1, 1, 0.25, 0.63);
+
+	int border_width = Math::round(1 * EDSCALE);
+
+	Rect2 rect = container->_edit_get_rect();
+
+	switch (type) {
+		case CONTAINER_TYPE_NONE:
+			break;
+		case CONTAINER_TYPE_ASPECT_RATIO: {
+			AspectRatioContainer *aspect_ratio_container = Object::cast_to<AspectRatioContainer>(container);
+			Size2 child_size = Size2(aspect_ratio_container->get_ratio(), 1.0);
+			float scale_factor = 1.0;
+
+			switch (aspect_ratio_container->get_stretch_mode()) {
+				case AspectRatioContainer::STRETCH_WIDTH_CONTROLS_HEIGHT: {
+					scale_factor = rect.size.x / child_size.x;
+				} break;
+				case AspectRatioContainer::STRETCH_HEIGHT_CONTROLS_WIDTH: {
+					scale_factor = rect.size.y / child_size.y;
+				} break;
+				case AspectRatioContainer::STRETCH_FIT: {
+					scale_factor = MIN(rect.size.x / child_size.x, rect.size.y / child_size.y);
+				} break;
+				case AspectRatioContainer::STRETCH_COVER: {
+					scale_factor = MAX(rect.size.x / child_size.x, rect.size.y / child_size.y);
+				} break;
+			}
+			child_size *= scale_factor;
+
+			float align_x = 0.5;
+			switch (aspect_ratio_container->get_alignment_horizontal()) {
+				case AspectRatioContainer::ALIGNMENT_BEGIN: {
+					align_x = 0.0;
+				} break;
+				case AspectRatioContainer::ALIGNMENT_CENTER: {
+					align_x = 0.5;
+				} break;
+				case AspectRatioContainer::ALIGNMENT_END: {
+					align_x = 1.0;
+				} break;
+			}
+			float align_y = 0.5;
+			switch (aspect_ratio_container->get_alignment_vertical()) {
+				case AspectRatioContainer::ALIGNMENT_BEGIN: {
+					align_y = 0.0;
+				} break;
+				case AspectRatioContainer::ALIGNMENT_CENTER: {
+					align_y = 0.5;
+				} break;
+				case AspectRatioContainer::ALIGNMENT_END: {
+					align_y = 1.0;
+				} break;
+			}
+			Vector2 offset = (rect.size - child_size) * Vector2(align_x, align_y);
+			Rect2 inner_rect = aspect_ratio_container->is_layout_rtl() ? Rect2(Vector2(rect.size.x - offset.x - child_size.x, offset.y), child_size) : Rect2(offset, child_size);
+
+			p_viewport_control->draw_rect(xform.xform(inner_rect), border_color, false, border_width);
+		} break;
+		case CONTAINER_TYPE_GRID: {
+			GridContainer *grid_container = Object::cast_to<GridContainer>(container);
+			for (int cell_index = 0; cell_index < grid_container->cell_sizes.size(); cell_index++) {
+				int32_t cell_offset = grid_container->cell_sizes[cell_index];
+				bool draw_row_lines = cell_index >= grid_container->get_columns() - 1;
+				Vector2 p1, p2;
```

Comment:
```suggestion
				Vector2 p1;
				Vector2 p2;
```
For all, one declaration per line please